common/middleware/ensure-current-location.js

```diff
@@ -1,9 +1,12 @@
 function ensureCurrentLocation({ locationsMountpath, whitelist = [] } = {}) {
   return (req, res, next) => {
+    const { permissions = [], locations = [] } = req.session.user || {}
+
     if (
       req.session.currentLocation ||
       whitelist.includes(req.url) ||
-      req.url.includes(locationsMountpath)
+      req.url.includes(locationsMountpath) ||
+      (permissions.includes('moves:view:all') && !locations.length)
```

Comment:
Maybe capturing this in a const would aid in explanation:

```
const supplierWithoutLocations = (permissions.includes('moves:view:all') && !locations.length)

```

or

```
const viewAllMovesWithoutLocations = (permissions.includes('moves:view:all') && !locations.length)

```

But then maybe it's not needed? I think it will become apparent after writing to see if it's too much. Your call and as mentioned after a chat and seeing the tests its become clear what it does.

Reply:
Was thinking about it over lunch and this would be the most descriptive: `canViewAllMovesWithoutLocations`...but is that too verbose?